packages/state-transition/src/block/processConsolidationRequest.ts

```diff
@@ -0,0 +1,72 @@
+import {electra, ssz} from "@lodestar/types";
+import {FAR_FUTURE_EPOCH, MIN_ACTIVATION_BALANCE, PENDING_CONSOLIDATIONS_LIMIT} from "@lodestar/params";
+
+import {CachedBeaconStateElectra} from "../types.js";
+import {getConsolidationChurnLimit, isActiveValidator} from "../util/validator.js";
+import {hasExecutionWithdrawalCredential} from "../util/electra.js";
+import {computeConsolidationEpochAndUpdateChurn} from "../util/epoch.js";
+
+export function processConsolidationRequest(
+  state: CachedBeaconStateElectra,
+  consolidationRequest: electra.ConsolidationRequest
+): void {
+  // If the pending consolidations queue is full, consolidation requests are ignored
+  if (state.pendingConsolidations.length >= PENDING_CONSOLIDATIONS_LIMIT) {
+    return;
+  }
+
+  // If there is too little available consolidation churn limit, consolidation requests are ignored
+  if (getConsolidationChurnLimit(state) <= MIN_ACTIVATION_BALANCE) {
+    return;
+  }
+
+  const {sourcePubkey, targetPubkey} = consolidationRequest;
+  const sourceIndex = state.epochCtx.getValidatorIndex(sourcePubkey);
+  const targetIndex = state.epochCtx.getValidatorIndex(targetPubkey);
+
+  if (sourceIndex === undefined || targetIndex === undefined) {
+    return;
+  }
+
+  // Verify that source != target, so a consolidation cannot be used as an exit.
+  if (sourceIndex === targetIndex) {
+    return;
+  }
+
+  const sourceValidator = state.validators.getReadonly(sourceIndex);
+  const targetValidator = state.validators.getReadonly(targetIndex);
+  const sourceWithdrawalAddress = sourceValidator.withdrawalCredentials.subarray(12);
+  const currentEpoch = state.epochCtx.epoch;
+
+  // Verify withdrawal credentials
+  if (
+    !hasExecutionWithdrawalCredential(sourceValidator.withdrawalCredentials) ||
+    !hasExecutionWithdrawalCredential(targetValidator.withdrawalCredentials)
+  ) {
+    return;
+  }
+
+  if (Buffer.compare(sourceWithdrawalAddress, consolidationRequest.sourceAddress) !== 0) {
+    return;
+  }
+
+  // Verify the source and the target are active
+  if (!isActiveValidator(sourceValidator, currentEpoch) || !isActiveValidator(targetValidator, currentEpoch)) {
+    return;
+  }
+
+  // Verify exits for source and target have not been initiated
+  if (sourceValidator.exitEpoch !== FAR_FUTURE_EPOCH || targetValidator.exitEpoch !== FAR_FUTURE_EPOCH) {
+    return;
+  }
+
+  const exitEpoch = computeConsolidationEpochAndUpdateChurn(state, BigInt(sourceValidator.effectiveBalance));
```

Comment:
in state transition we already computed everything by EFFECTIVE_BALANCE_INCREMENT to get rid of BigInt to improve performance
there's a chance that we can do that too with this computation, need to explore that later
could you mark a TODO so that we don't miss this?